src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java

```diff
@@ -365,6 +369,18 @@ private void startTriggerThreads() {
         return null;
     }
 
+    public void setJobAction(Exception e) {
+        action = new TriggerThreadProblemAction(e);
+    }
+
+    public Action getJobAction() {
```

Comment:
Consider either returning explicit type (`TriggerThreadProblemAction`) or a collection of actions. It has 2 advantages:

- encapsulates `cibt.getJobAction() == null ? Collections.emptyList() : Collections.singletonList(cibt.getJobAction())` from `TriggerThreadTransientActionFactory` (the code is now thread unsafe, fwiw).
- get the api ready for adding more than one action - hypothetical scenario really.

Reply:
Done.